OpenXmlPowerTools/PtOpenXmlUtil.cs

```diff
@@ -743,66 +745,42 @@ public static int CalcWidthOfRunInTwips(XElement r)
                 try
                 {
-                    var fs2 = FontStyle.Regular;
-                    using (Font f = new Font(ff, (float)sz / 2f, fs2))
-                    {
-                        System.Windows.Forms.TextFormatFlags tff = System.Windows.Forms.TextFormatFlags.NoPadding;
-                        Size proposedSize = new Size(int.MaxValue, int.MaxValue);
-                        var sf = System.Windows.Forms.TextRenderer.MeasureText(runText, f, proposedSize, tff); // sf returns size in pixels
-                        var dpi = 96m;
-                        var twip = (int)(((sf.Width / dpi) * 1440m) / multiplier + tabLength * 1440m);
-                        return twip;
-                    }
+                    w = MetricsGetter.GetTextWidth(ff, fs2, sz, runText);
                 }
                 catch (ArgumentException)
                 {
-                    var fs2 = FontStyle.Bold;
                     try
                     {
-                        using (Font f = new Font(ff, (float)sz / 2f, fs2))
-                        {
-                            System.Windows.Forms.TextFormatFlags tff = System.Windows.Forms.TextFormatFlags.NoPadding;
-                            Size proposedSize = new Size(int.MaxValue, int.MaxValue);
-                            var sf = System.Windows.Forms.TextRenderer.MeasureText(runText, f, proposedSize, tff); // sf returns size in pixels
-                            var dpi = 96m;
-                            var twip = (int)(((sf.Width / dpi) * 1440m) / multiplier + tabLength * 1440m);
-                            return twip;
-                        }
+                        w = MetricsGetter.GetTextWidth(ff, fs3, sz, runText);
                     }
                     catch (ArgumentException)
                     {
                         // if both regular and bold fail, then get metrics for Times New Roman
                         // use the original FontStyle (in fs)
-                        FontFamily ff2;
-                        ff2 = new FontFamily("Times New Roman");
-                        using (Font f = new Font(ff2, (float)sz / 2f, fs))
-                        {
-                            System.Windows.Forms.TextFormatFlags tff = System.Windows.Forms.TextFormatFlags.NoPadding;
-                            Size proposedSize = new Size(int.MaxValue, int.MaxValue);
-                            var sf = System.Windows.Forms.TextRenderer.MeasureText(runText, f, proposedSize, tff); // sf returns size in pixels
-                            var dpi = 96m;
-                            var twip = (int)(((sf.Width / dpi) * 1440m) / multiplier + tabLength * 1440m);
-                            return twip;
-                        }
+                        w = MetricsGetter.GetTextWidth(ff2, fs, sz, runText);
                     }
                 }
             }
+            catch (OverflowException)
+            {
+                return 0;
```

Comment:
`MetricsGetter....` should handle all exceptions

Reply:
I'll remove this catch because anyways GetTextWidth cacthes everything.